### PR TITLE
unit tests: add coverage for unset

### PR DESCRIPTION
**Pull request by duthils**

### CODE REVIEW

File: sops_test.go [hunk 1]
```diff
@@ -1108,6 +1108,322 @@ func TestSetArrayNonLeaf(t *testing.T) {
...
+					Value: "bar",
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", "unknown"})
+	assert.Equal(t, err.(*SopsKeyNotFound).Key, "unknown")
+	if unset != nil {
+		t.Errorf("Unset result was not nil upon %s", err)
+	}
```
Comment (felixfontein) [inline sops_test.go:1368]: I think you can also simply do
```suggestion
	assert.Nil(t, unset, "Unset result was not nil upon %s", err)
```
(https://github.com/stretchr/testify/blob/b074924938f86d417f1c9a845c7e8b0784d7f937/assert/assertions.go#L704-L715)

Same below.

File: sops_test.go [hunk 2]
```diff
@@ -1108,6 +1108,322 @@ func TestSetArrayNonLeaf(t *testing.T) {
...
+					TreeItem{
+						Key: "bar",
+						Value: "bar",
+					},
+				},
+			},
+		},
+	}
+	unset, err := branch.Unset([]interface{}{"foo", 0})
+	assert.Equal(t, nil, err)
```
Comment (felixfontein) [inline sops_test.go:1342]:
```suggestion
	assert.Nil(t, err)
```

**Review by felixfontein — COMMENTED** [review]
Thanks for improving the tests!

File: sops_test.go (hunk 1 above)
Comment (duthils) [reply sops_test.go]: Thanks for the tip!

File: sops_test.go (hunk 2 above)
Comment (duthils) [reply sops_test.go]: Thanks for the tip!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@duthils thanks for your contribution!

**Pull request merged**